core/trino-spi/src/main/java/io/trino/spi/connector/DropSuccess.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+public final class DropSuccess
```

Comment:
You can DropResult an interface

```java
public sealed interface DropResult
        permits DropSuccess, DropRedirected
```

and then this can be plain enum:

```java
public enum DropSuccess
        implements DropResult
{
    INSTANCE;

    @Override
    public boolean isRedirected()
    {
        return false;
    }
}
```

**HOWEVER** the more i think about this the more i feel it's better to have a flat class

```java
public final class DropResult
{
    public static DropResult success()
    {
        return new DropResult(Optional.empty());
    }

    public static DropResult redirected(CatalogSchemaTableName target)
    {
        return new DropResult(Optional.of(requireNonNull(target, "target is null")));
    }

    private final Optional<CatalogSchemaTableName> redirected;

    private DropResult(Optional<CatalogSchemaTableName> redirected)
    {
        this.redirected = requireNonNull(redirected, "redirected is null");
    }

    public boolean isSuccess()
    {
        return redirected.isEmpty();
    }

    public boolean isRedirected()
    {
        return redirected.isPresent();
    }

    public CatalogSchemaTableName getRedirectionTarget()
    {
        return redirected.orElseThrow(() -> new UnsupportedOperationException("Not redirected"));
    }

    @Override
    public boolean equals(Object o)
    {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        DropResult that = (DropResult) o;
        return Objects.equals(redirected, that.redirected);
    }

    @Override
    public int hashCode()
    {
        return redirected.hashCode();
    }

    @Override
    public String toString()
    {
        if (isSuccess()) {
            return "DropResult.success()";
        }
        return "DropResult.redirected(%s)".formatted(getRedirectionTarget());
    }
}
```